test/cli/cli_test.cpp

```diff
@@ -45,9 +46,33 @@ struct cli_test : public ::testing::Test
 
     // Generate the full path of a test file that is provided in the data directory.
     static
-    std::string data(std::string const & filename)
+    std::filesystem::path data(std::string const & filename)
     {
-        return std::string{DATADIR}.append(filename);
+        return std::filesystem::path{std::string{DATADIR}}.concat(filename);
+    }
+
+    // Create a unique work directory for a test and return the full path for a given file name.
+    static
+    std::filesystem::path workdir(std::string const & filename)
+    {
+        // Derive unique test directory name from the test name.
+        ::testing::TestInfo const * const info = ::testing::UnitTest::GetInstance()->current_test_info();
+        std::string test_name = std::string{info->test_case_name()} + std::string{"."} + std::string{info->name()};
+        std::filesystem::path test_dir{std::string{OUTPUTDIR} + test_name};
+
+        try // Create new directory (if it does not exist).
+        {
+            std::filesystem::create_directories(test_dir);
+        }
+        catch (std::filesystem::filesystem_error err)
+        {
+            // fall back to current directory
+            std::cerr << err.what() << std::endl;
+            return std::filesystem::path{test_name + filename};
```

Comment:
Why should this catch case work? What happens if the folder "test_name" does not exist?

```suggestion
            test_dir = std::filesystem::path{test_name};
```

Reply:
It seems that my naming is not good, because you misunderstood what it is doing. If we arrive in the catch statement, then `test_dir` could not be created (if disk is full or missing write permission). Then I create an output file with the (unique) name <test_name>out.fasta in the current work directory, if the filename was "out.fasta.

Reply:
I think the test should just fail with an exception in this case. This is a test and the test infrastructure should work and if it doesn't it should clearly tell you

Reply:
Sure, then I just delete the catch block and the test will throw exceptions in that case.